CMakeLists.txt

```diff
@@ -788,24 +820,31 @@ if(NRN_ENABLE_DOCS)
     WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/docs
     COMMENT "Generating documentation with Sphinx")
 
-  # Executing notebooks will add outputs in-place. We don't want those committed to the repo. This
-  # commands cleans them out.
-  add_custom_target(
-    notebooks-clean
-    COMMAND ${NRN_DOCS_COMMAND_PREFIX} bash ${PROJECT_SOURCE_DIR}/docs/notebooks.sh --clean
-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/docs)
+  # we need to first run the notebooks, and _then_ build the documentation via sphinx
+  add_dependencies(sphinx ${all_notebooks} doxygen)
+
+  if(NRN_ENABLE_DOCS_WITH_EXTERNAL_INSTALLATION)
+    message(STATUS "**Not** making the notebooks target depend on the rest of the NEURON build.")
+    message(STATUS "Documentation building will probably fail if you haven't installed NEURON.")
+  else()
+    # We need NEURON and NMODL to execute the notebooks. If we're building documentation as part of
+    # a normal build, this means we need to schedule the documentation targets sufficiently late in
+    # the build.
+    foreach(notebook ${all_notebooks})
+      add_dependencies(${notebook} _nmodl hoc_module rxdmath nrnivmodl_dependency)
```

Comment:
The notebooks don't actually need the `_nmodl` target yet (see comment above about NMODL notebooks), but since readthedocs runs sphinx on its own, we still need to build the NMODL Python module beforehand, otherwise the Python docs will be missing.